client.go

```diff
@@ -1,148 +1,192 @@
+    rdr, err := zlib.NewReader(bytes.NewBuffer(*packet))
+    if err != nil {
+        log.WithFields(log.Fields{
+            "error": err,
+        }).Fatal("Decompress error")
+        return Message{}, err
+    }
+    var out bytes.Buffer
+    io.Copy(&out, rdr)
+    rdr.Close()
+    decompressedRaw := out.Bytes()
+    decompressed := decompressedRaw[DecompressedHeaderIndex:]
+    var m Message
+    err = json.Unmarshal(decompressed, &m)
+    if err != nil {
+        log.WithFields(log.Fields{
+            "error": err,
+        }).Fatal("Unmarshal error")
+        return Message{}, err
+    }
+    return m, nil
+}
+
+// SendMessage sends a packet to another client based on destination data
+func (c *Client) SendMessage(m *Message, packet *[]byte) {
+    sent := make([]string, 0)
+    for _, to := range m.To {
+        if sender, ok := c.hub.clientMap[c.privateKey][to]; ok {
+            exists := false
+            for _, check := range sent {
+                if check == to {
+                    exists = true
+                    break
+                }
+            }
+            if !exists {
+                sent = append(sent, to)
+                snd := SendPacket{addr: sender.addr, data: append([]byte{}, *packet...)}
+                select {
+                case c.hub.sendChannel <- snd:
+                default:
+                }
+            }
+        } else { // Regex
+            for name, sender := range c.hub.clientMap[c.privateKey] {
+                if name != m.From {
+                    match, _ := regexp.MatchString(to, name)
+                    if match {
+                        exists := false
+                        for _, check := range sent {
+                            if check == name {
+                                exists = true
+                                break
+                            }
+                        }
+                        if !exists {
+                            sent = append(sent, name)
+                            snd := SendPacket{addr: sender.addr, data: append([]byte{}, *packet...)}
+                            select {
+                            case c.hub.sendChannel <- snd:
+                            default:
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+// ProcessDescription processes a description message and updates client descriptions
+func (c *Client) ProcessDescription(m *Message, clientName string) error {
+    var desc Description
+    err := json.Unmarshal(m.Msg, &desc)
+    if err != nil {
+        log.WithFields(log.Fields{
+            "error": err,
+        }).Fatal("Unmarshal error")
+        return err
+    }
+    if sender, ok := c.hub.clientMap[c.privateKey][clientName]; ok {
+        sender.description = desc.Data
+    }
+    return nil
+}
+
+func processTopicString(topic string) (string, string) {
+    splitTopic := strings.Split(topic, "/")
```

Comment:
Would be nice to add some assert statements indicating the expected formatting for `topic`. This would be better than receiving arbitrary "index out of bounds" errors on the topicName and clientName parsing.

Btw I don't think the Go standard library comes with asserts, so instead you can just `panic()` if certain conditions aren't met.